### PR TITLE
Improve RosParameterHandler

### PR DESCRIPTION
**Pull request by hslusarek**
This PR  removes code smells from the `RosParameterHandler` and simplifies it.

### CODE REVIEW

File: src/psen_scan_driver.cpp [hunk 1]
```diff
@@ -52,17 +67,19 @@ int main(int argc, char** argv)
...
+        getRequiredParamFromServer<int>(pnh, PARAM_HOST_CONTROL_PORT),
+        getRequiredParamFromServer<std::string>(pnh, PARAM_SCANNER_IP),
+        DEFAULT_X_AXIS_ROTATION - getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_END, DEFAULT_ANGLE_END),
+        DEFAULT_X_AXIS_ROTATION - getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_START, DEFAULT_ANGLE_START));
+
+    ROSScannerNode ros_scanner_node(pnh,
+                                    DEFAULT_PUBLISH_TOPIC,
+                                    getOptionalParamFromServer<std::string>(pnh, PARAM_FRAME_ID, DEFAULT_FRAME_ID),
+                                    DEFAULT_X_AXIS_ROTATION,
+                                    scanner_configuration);
```
Comment (martiniil) [inline src/psen_scan_driver.cpp:82]:
```suggestion
    const auto host_ip{ getRequiredParamFromServer<std::string>(pnh, PARAM_HOST_IP) };
    const auto host_data_port{ getRequiredParamFromServer<int>(pnh, PARAM_HOST_DATA_PORT) };
    const auto host_control_port{ getRequiredParamFromServer<int>(pnh, PARAM_HOST_CONTROL_PORT) };
    const auto scanner_ip{ getRequiredParamFromServer<std::string>(pnh, PARAM_SCANNER_IP) };
    const auto angle_start{ DEFAULT_X_AXIS_ROTATION -
                            getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_END, DEFAULT_ANGLE_END) };
    const auto angle_end{ DEFAULT_X_AXIS_ROTATION -
                          getOptionalParamFromServer<double>(pnh, PARAM_ANGLE_START, DEFAULT_ANGLE_START) };

    ScannerConfiguration scanner_configuration(
        host_ip, host_data_port, host_control_port, scanner_ip, angle_start, angle_end);

    const auto frame_id{ getOptionalParamFromServer<std::string>(pnh, PARAM_FRAME_ID, DEFAULT_FRAME_ID) };

    ROSScannerNode ros_scanner_node(
        pnh, DEFAULT_PUBLISH_TOPIC, frame_id, DEFAULT_X_AXIS_ROTATION, scanner_configuration);
```

Comment (martiniil) [reply src/psen_scan_driver.cpp:82]: I was wondering if the readability is better if the parameters or initialized on separate lines. What do you think. At least concerning the start-/end-angle I would prefer it like that.

**Review by martiniil — APPROVED** [review]
One comment about code readability. Decide for yourself.

File: src/psen_scan_driver.cpp (hunk 1 above)
Comment (hslusarek) [reply src/psen_scan_driver.cpp]: I opted for a compromise: I initialized the parameters start- and end-angle on a separate line.

**Pull request merged**